### ISSUE TITLE
dotnet new mvc experiance on macos seems broken - problems window full of issues

### ISSUE THREAD

**Issue by solrevdev** [OmniSharp/omnisharp-vscode#3477]
Please paste the output from your clipboard## Issue Description ##
## Steps to Reproduce ##

For me, running `dotnet new mvc` then `code .` in the terminal followed by `dotnet build` in VSCode's terminal window compiled fine but showed approx 120 problems in VSCode's problems window. 

Red squiggles everywhere!

For more info and screenshots see this twitter thread [https://twitter.com/solrevdev/status/1207634539469819904](https://twitter.com/solrevdev/status/1207634539469819904)

## Expected Behavior ##

Problems window in VSCode should have no errors

## Actual Behavior ##

Problems window in VSCode full of errors

### OmniSharp log ###
<details>
Starting OmniSharp server at 20/12/2019, 09:14:21
    Target: /Users/solrevdev/Dropbox/Projects/example code/dropbox/MvcApp

OmniSharp server started with Mono 6.6.0.
    Path: /Users/solrevdev/.vscode-insiders/extensions/ms-vscode.csharp-1.21.9/.omnisharp/1.34.9/omnisharp/OmniSharp.exe
    PID: 61325

[info]: OmniSharp.Stdio.Host
        Starting OmniSharp on MacOS 10.14.6 (x64)
[info]: OmniSharp.Services.DotNetCliService
        DotNetPath set to dotnet
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Located 2 MSBuild instance(s)
            1: Mono 16.3 - "/usr/local/Cellar/mono/6.6.0.161/lib/mono/msbuild/Current/bin"
            2: StandAlone 16.3 - "/Users/solrevdev/.vscode-insiders/extensions/ms-vscode.csharp-1.21.9/.omnisharp/1.34.9/omnisharp/.msbuild/Current/Bin"
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Registered MSBuild instance: Mono 16.3 - "/usr/local/Cellar/mono/6.6.0.161/lib/mono/msbuild/Current/bin"
            CscToolExe = csc.exe
            CscToolPath = /Users/solrevdev/.vscode-insiders/extensions/ms-vscode.csharp-1.21.9/.omnisharp/1.34.9/omnisharp/.msbuild/Current/Bin/Roslyn
[info]: OmniSharp.Cake.CakeProjectSystem
        Detecting Cake files in '/Users/solrevdev/Dropbox/Projects/example code/dropbox/MvcApp'.
[info]: OmniSharp.Cake.CakeProjectSystem
        Could not find any Cake files
[info]: OmniSharp.MSBuild.ProjectSystem
        No solution files found in '/Users/solrevdev/Dropbox/Projects/example code/dropbox/MvcApp'
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for '/Users/solrevdev/Dropbox/Projects/example code/dropbox/MvcApp/MvcApp.csproj'
[info]: OmniSharp.Script.ScriptProjectSystem
        Detecting CSX files in '/Users/solrevdev/Dropbox/Projects/example code/dropbox/MvcApp'.
[info]: OmniSharp.Script.ScriptProjectSystem
        Could not find any CSX files
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.CSharpFormattingWorkspaceOptionsProvider, Order: 0
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: /Users/solrevdev/Dropbox/Projects/example code/dropbox/MvcApp/MvcApp.csproj
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.RenameWorkspaceOptionsProvider, Order: 100
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.EditorConfigWorkspaceOptionsProvider, Order: 200
[info]: OmniSharp.WorkspaceInitializer
        Configuration finished.
[info]: OmniSharp.Stdio.Host
        Omnisharp server running using Stdio at location '/Users/solrevdev/Dropbox/Projects/example code/dropbox/MvcApp' on host 61053.
[info]: OmniSharp.MSBuild.ProjectManager
        Successfully loaded project file '/Users/solrevdev/Dropbox/Projects/example code/dropbox/MvcApp/MvcApp.csproj'.
[info]: OmniSharp.MSBuild.ProjectManager
        Adding project '/Users/solrevdev/Dropbox/Projects/example code/dropbox/MvcApp/MvcApp.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Update project: MvcApp
[warn]: OmniSharp.MSBuild.PackageDependencyChecker
        MvcApp: Did not find 'Microsoft.NETCore.App' in lock file.
[info]: OmniSharp.Roslyn.CSharp.Services.Diagnostics.CSharpDiagnosticWorkerWithAnalyzers
        Solution initialized -> queue all documents for code analysis. Initial document count: 6.

</details>

### C# log ###
<details>Emty</details>

## Environment information ##

**VSCode version**: 1.42.0-insider
**C# Extension**: 1.21.9

<details><summary>Mono Information</summary>
        OmniSharp using global mono :6.6.0</details>
<details><summary>Dotnet Information</summary>
.NET Core SDK (reflecting any global.json):
 Version:   3.1.100
 Commit:    cd82f021f4

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.14
 OS Platform: Darwin
 RID:         osx.10.14-x64
 Base Path:   /usr/local/share/dotnet/sdk/3.1.100/

Host (useful for support):
  Version: 3.1.0
  Commit:  65f04fb6db

.NET Core SDKs installed:
  3.1.100 [/usr/local/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.1.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.13 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|
|---|---|---|
|advanced-new-file|patbenatar|1.2.0|
|arc-plus|ph-hawkins|1.0.1|
|auto-close-tag|formulahendry|0.5.6|
|auto-rename-tag|formulahendry|0.1.1|
|azure-account|ms-vscode|0.8.7|
|bash-debug|rogalmic|0.3.6|
|bash-ide-vscode|mads-hartmann|1.4.0|
|beautify|HookyQR|1.5.0|
|bracket-pair-colorizer|CoenraadS|1.0.61|
|chat|karigari|0.31.0|
|code-settings-sync|Shan|3.4.3|
|code-spell-checker|streetsidesoftware|1.7.20|
|code-streaming|gitduck|0.0.33|
|csharp|ms-vscode|1.21.9|
|csharpextensions|jchannon|1.3.1|
|dark-plus-material|vangware|2.2.1|
|debugger-for-chrome|msjsdiag|4.12.3|
|docomment|k--kato|0.1.8|
|dotenv|mikestead|1.0.1|
|dotnet-test-explorer|formulahendry|0.7.1|
|EditorConfig|EditorConfig|0.14.4|
|editorconfiggenerator|nepaul|0.2.1|
|emojisense|bierner|0.5.3|
|es7-react-js-snippets|dsznajder|2.5.0|
|gc-excelviewer|GrapeCity|2.1.32|
|gistfs|vsls-contrib|0.0.23|
|githistory|donjayamanne|0.4.11|
|github-plus-theme|thenikso|1.2.1|
|gitlens|eamodio|10.2.0|
|horizon-theme-vscode|jolaleye|2.0.2|
|html-css-class-completion|Zignd|1.19.0|
|html-snippets|abusaidm|0.2.1|
|identical-sublime-monokai-csharp-theme-colorizer|maximetinu|0.11.1|
|linux-themes|SolarLiner|1.0.1|
|LiveServer|ritwickdey|5.6.1|
|LogFileHighlighter|emilast|2.6.0|
|markdown-all-in-one|yzhang|2.6.1|
|material-icon-theme|PKief|3.9.2|
|material-palenight-theme|whizkydee|1.9.7|
|material-theme|zhuangtongfa|3.2.1|
|msbuild-project-tools|tintoy|0.3.2|
|night-owl|sdras|1.1.3|
|noctis|liviuschera|10.22.0|
|npm-intellisense|christian-kohler|1.3.0|
|nuget-reverse-package-search|jesschadwick|0.1.68|
|one-monokai|azemoh|0.4.4|
|partial-diff|ryu1kn|1.4.0|
|path-intellisense|christian-kohler|1.4.2|
|polacode|pnp|0.3.4|
|powershell|ms-vscode|2019.12.0|
|prettier-vscode|esbenp|3.17.0|
|python|ms-python|2019.12.54270-dev|
|quicktype|quicktype|12.0.46|
|quokka-vscode|WallabyJs|1.0.267|
|razor-snippets-for-asp-net-core|aledev29|0.0.3|
|regex|chrmarti|0.2.0|
|remote-containers|ms-vscode-remote|0.96.0|
|remote-ssh|ms-vscode-remote|0.48.0|
|remote-ssh-edit|ms-vscode-remote|0.48.0|
|remote-ssh-explorer|ms-vscode-remote|0.48.0|
|remote-wsl|ms-vscode-remote|0.41.6|
|rest-client|humao|0.23.0|
|roslynator|josefpihrt-vscode|2.2.0|
|shell-format|foxundermoon|6.1.3|
|sort-lines|Tyriar|1.9.0|
|sqltools|mtxr|0.21.5|
|swagger-viewer|Arjun|2.2.2|
|theme-cobalt2|wesbos|2.1.6|
|theme-dracula|dracula-theme|2.19.0|
|theme-monokai-pro-vscode|monokai|1.1.14|
|theme-panda|tinkertrain|1.3.0|
|theme-setimonokai|SmukkeKim|0.0.1|
|transformer|dakara|1.6.0|
|vetur|octref|0.22.6|
|viml|dunstontc|0.1.7|
|vsc-material-theme|Equinusocio|30.0.0|
|vscode-css-peek|pranaygp|3.0.2|
|vscode-custom-css|be5invis|3.0.9|
|vscode-docker|ms-azuretools|0.9.0|
|vscode-eslint|dbaeumer|2.0.6|
|vscode-firefox-debug|firefox-devtools|2.6.1|
|vscode-html-css|ecmel|0.2.3|
|vscode-icons|vscode-icons-team|9.6.0|
|vscode-language-pack-en-GB|MS-CEINTL|1.41.2|
|vscode-mysql|formulahendry|0.4.0|
|vscode-npm-script|eg2|0.3.10|
|vscode-nuget-package-manager|jmrog|1.1.6|
|vscode-openapi|42Crunch|2.0.2|
|vscode-peacock|johnpapa|3.2.0|
|vscode-remote-extensionpack|ms-vscode-remote|0.19.0|
|vscode-sort-json|richie5um2|1.18.0|
|vscode-surround|yatki|1.0.2|
|vscode-theme-onedark|akamud|2.1.0|
|vscode-theme-onelight|akamud|2.1.0|
|vscode-todo-plus|fabiospampinato|4.14.1|
|vscode-trello-viewer|Ho-Wan|0.6.0|
|vscode-typescript-tslint-plugin|ms-vscode|1.2.3|
|vscode-wakatime|WakaTime|2.2.1|
|vscode-yaml|redhat|0.6.1|
|vscodeintellicode|VisualStudioExptTeam|1.2.2|
|vsliveshare|ms-vsliveshare|1.0.1404|
|vsliveshare-audio|ms-vsliveshare|0.1.80|
|vsliveshare-pack|ms-vsliveshare|0.3.4|
|vsonline|ms-vsonline|1.0.1424|
|winteriscoming|johnpapa|1.3.0|
|wrapSelection|konstantin|0.6.8|
|xml|DotJoshJohnson|2.5.0|;

</details>

**Comment by filipw** [comment]
looks like the problem is i nthe Cellar installation of Mono.
Try installing mono 6.6.0 from the official installer https://www.mono-project.com/download/stable/ or set `"omnisharp.useGlobalMono": "never"`

**Comment by solrevdev** [comment]
Success! 🎉

Thank you @filipw that was indeed the issue! 

So for completeness, these are the steps I took. 

I [downloaded version 6.4.0.198](https://download.mono-project.com/archive/6.4.0/macos-10-universal/MonoFramework-MDK-6.4.0.198.macos10.xamarin.universal.pkg) from the mono site (as I have Visual Studio for macOS also installed) and ran the installer. 

I then ran `source .zshrc` from my terminal to refresh things followed by `mono --version`

So this is the version now installed globally.

![Screenshot 2019-12-20 09 51 01](https://user-images.githubusercontent.com/374863/71246420-82d91300-230e-11ea-909b-742b0229baee.png)

I then restarted VSCode and  🎉 

![Screenshot 2019-12-20 09 50 48](https://user-images.githubusercontent.com/374863/71246419-82d91300-230e-11ea-8f81-1dc1031fe4e1.png)
